src/fileformats/caoslexer.cpp

```diff
@@ -205,6 +205,8 @@ void lexcaos(std::vector<caostoken>& v, const char* p) {
 		push_value(caostoken::TOK_ERROR);
 	} else if (p[0] == '\r' || p[0] == '\n') {
 		p++;
+		yylineno++;
```

Comment:
Only `\n` (or `\r\n` if treated as an atomic unit) is a newline and should increment the line number. As is, this would cause an erroneous double line increment for `[\r\n`